src/app/CommandHandler.cpp

```diff
@@ -257,9 +258,10 @@ CHIP_ERROR CommandHandler::ProcessCommandDataIB(CommandDataIB::Parser & aCommand
     {
         Access::SubjectDescriptor subjectDescriptor = mpExchangeCtx->GetSessionHandle()->GetSubjectDescriptor();
         Access::RequestPath requestPath{ .cluster = concretePath.mClusterId, .endpoint = concretePath.mEndpointId };
-        Access::Privilege requestPrivilege = Access::Privilege::kOperate; // TODO: get actual request privilege
-        err                                = Access::GetAccessControl().Check(subjectDescriptor, requestPath, requestPrivilege);
-        err                                = CHIP_NO_ERROR; // TODO: remove override
+        Access::Privilege requestPrivilege =
+            Access::RequiredPrivilege::ForInvokeCommand(concretePath.mClusterId, concretePath.mEndpointId, concretePath.mCommandId);
```

Comment:
Is there a reason ForInvokeCommand does not just take a ConcreteCommandPath?

Reply:
That is simple. Those APIs exist at the "app" level while access control is a system level API. Eventually, we may wish to push all those path-like objects down to a more "core" level, but that hasn't been done.

Arguably, we could put this whole RequiredPrivilege module higher, in the "app" level, since it isn't directly interacting with other aspects of access control. I put it here for now, and wasn't worried about a few args. If we feel this isn't going to be so temporary, I could move it into app, if we prefer it there. I just worry app is becoming a default place for everything, due to less than desirable layering in the system.

Reply:
I moved the class up into app, and changed them to take those ConcreteBlahPath objects.